test/test_namedtensor.py

```diff
@@ -272,6 +272,19 @@ def test_diagonal(self):
         self.assertEqual(named_tensor.diagonal(outdim='E', dim1='B', dim2='D').names,
                          ['A', 'C', 'E'])
 
+    def test_max_polling(self):
+        named_tensor_1d = torch.zeros(2, 3, 5, names=list('ABC'))
+        named_tensor_2d = torch.zeros(2, 3, 5, 7, names=list('ABCD'))
+        named_tensor_3d = torch.zeros(2, 3, 5, 7, 9, names=list('ABCDE'))
+
+        self.assertEqual(named_tensor_1d.names, F.max_pool1d(named_tensor_1d, 2).names)
+        self.assertEqual(named_tensor_2d.names, F.max_pool2d(named_tensor_2d, [2, 2]).names)
+        self.assertEqual(named_tensor_3d.names, F.max_pool3d(named_tensor_3d, [2, 2, 2]).names)
+
+        self.assertEqual(named_tensor_1d.names, F.max_pool1d_with_indices(named_tensor_1d, 2)[0].names)
```

Comment:
The indices should have names as well. Because the indices have the same shape as the result, the indices should probably have the same names as the result.

i.e., if our input has names `list("NCHW")`, then the output has names `list("NCHW")` and the indices should also have names `list("NCHW")` because the data has the same dimension ordering.

Action item: we should propagate names to the indices and test that they have the same names.